### PR TITLE
Ts2p8 upgrade

### PR DESCRIPTION
**Pull request by Bnaya**
Simply upgrade typescript to 2.8
First step toward improving types using 2.8 features
ref #705

### CODE REVIEW

File: packages/mst-middlewares/src/time-traveller.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
     ISnapshottable,
     IModelType
 } from "mobx-state-tree"
+import { IObservableArray } from "mobx"
```
Comment (Bnaya) [inline packages/mst-middlewares/src/time-traveller.ts]: without it, i got this error:
```
src/time-traveller.ts(13,7): error TS4023: Exported variable 'TimeTraveller' has or is using name 'IObservableArray' from external module "/Users/bnaya/dev/opensource/mobx-state-tree/node_modules/mobx/lib/types/observablearray" but cannot be named.
```

**Comment by Bnaya** [comment]
ref #705

**Comment by Bnaya** [comment]
@robinfehr should i go on and merge?

**Comment by mweststrate** [comment]
@Bnaya / @robinfehr yeah feel free to merge reviewed PR's that don't overhaul the package :)

**Comment by mweststrate** [comment]
Ehh but, no `package.json` config changes? 

(lol, I already merged :), see don't trust me)

**Pull request merged**